README.md

```diff
@@ -1 +1,76 @@
-# modelplace
\ No newline at end of file
+This is an instruction on how to integrate a model to [Modelplace.AI](https://modelplace.ai/).  
+We support [PyTorch](https://pytorch.org/), [Tensorflow](https://www.tensorflow.org/), [OpenVINO](https://docs.openvinotoolkit.org/latest/index.html), and [OnnxRuntime](https://www.onnxruntime.ai/) backends. 
+
+## Publishing information
+For publishing you must provide information about yourself and the model
+### Required
+**Model description** - this is the whole description of what the model does (e.g. [model page](https://modelplace.ai/models/2) - the text in Summary block).
+
+**Model description preview short** - the text people see on a card on the [main page](https://modelplace.ai/) (just after the model name)
+
+**Model mame full** - on a model page - (e.g. Faster region-based convolutional neural network with ResNet-50 FPN backbone https://modelplace.ai/models/2)
+
+**Model name short** - on the [main page](https://modelplace.ai/) (e.g. Faster R-CNN).
+
+**Dataset name** - it's may be public data e.g. MSCOCO or Open images, or your internal dataset.
+
+**Preview image** - we will run a model on the image and will use its output for visualization. This image will be both on the [main page](https://modelplace.ai/) and on the [model page](https://modelplace.ai/models/2)
+
+**Licence** - licence type e.g. Apache 2.0, MIT, Proprietary and so on
+**Number of classes** 
+**Author** - your name or company name
+**Metrics** - accuracy of your model
+
+### Optional
+**Dataset link** - link to dataset that you use for training and testing of your model
+**Source link** - link to your project page or profile page or company page and so on
+**System requirements** - what system setup( e.g. number of CPU GPU RAM) we should use to serve your model.
+
+**Logo** - we will add it to the preview.   
+_Note: This should be svg(png is possible) image with circle shape and size 128x128 pixels at least._
+
+**Avatar image** - you can add your own logo(by default we use our).   
+_Note: This should be svg(png is possible) image with circle shape and size 128x128 pixels at least._
+
+## Evironment
+- Install venv  
+```python3.7 -m pip install virtualenv```
+- Create an empty virtual environment for python3.7  
+```python3.7 -m  virtualenv  venv```
+- Activate it  
+```source venv/bin/activate```
+- Install pytest and wheel  
+```python3.7 -m pip install pytest wheel```
+- Install git and git-lfs  
+```sudo apt install git git-lfs```
+- Clone the repo 
+```git clone https://github.com/opencv-ai/modelplace.git```
+
+## Example
+You can run [an example](pytorch_fastercnn) of a package
+- Go to package folder pytorch_fastercnn
+```cd pytorch_fastercnn```
+- Install the package  
+```python3.7 setup.py install develop```
+- Run tests  
+```pytest pytorch_fastercnn```
```

Comment:
**Critical:**

`pytest tests` command must be run as there are no tests inside `pytorch_fastercnn` folder

Tried to run tests on MacBook, failed. What am I doing wrong?

Logs:
```
(env) lxdv@macbook:~/Desktop/modelplace/pytorch_fastercnn (vbebnev/manual)$ pytest tests/
============================= test session starts ==============================
platform darwin -- Python 3.7.10, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
rootdir: /Users/lxdv/Desktop/modelplace/pytorch_fastercnn
collected 3 items                                                              

tests/test_pytorch_fastercnn.py FF.                                      [100%]

=================================== FAILURES ===================================
_____________________________ test_process_sample ______________________________

prepare_data = (<PIL.Image.Image image mode=RGB size=640x425 at 0x7FDB47DB1C90>, [{'class_name': 'potted plant', 'score': 0.976785123...1': 404, 'x2': 433, ...}, {'class_name': 'potted plant', 'score': 0.9167777299880981, 'x1': 170, 'x2': 234, ...}, ...])

    def test_process_sample(prepare_data):
        test_image, test_result = prepare_data
    
        model = InferenceModel()
        model.model_load(BACKEND)
        ret = model.process_sample(test_image)
        ret = [pydantic.json.pydantic_encoder(item) for item in ret]
>       assert is_equal(ret, test_result)
E       AssertionError: assert False
E        +  where False = is_equal([{'class_name': 'potted plant', 'score': 0.9737666249275208, 'x1': 363, 'x2': 397, ...}, {'class_name': 'dining table'...77942, 'x1': 405, 'x2': 433, ...}, {'class_name': 'bowl', 'score': 0.9188650846481323, 'x1': 563, 'x2': 585, ...}, ...], [{'class_name': 'potted plant', 'score': 0.9767851233482361, 'x1': 363, 'x2': 398, ...}, {'class_name': 'chair', 'scor...x1': 404, 'x2': 433, ...}, {'class_name': 'potted plant', 'score': 0.9167777299880981, 'x1': 170, 'x2': 234, ...}, ...])

tests/test_pytorch_fastercnn.py:34: AssertionError
_____________________________ test_batch_inference _____________________________

prepare_data = (<PIL.Image.Image image mode=RGB size=640x425 at 0x7FDAFFC387D0>, [{'class_name': 'potted plant', 'score': 0.976785123...1': 404, 'x2': 433, ...}, {'class_name': 'potted plant', 'score': 0.9167777299880981, 'x1': 170, 'x2': 234, ...}, ...])

    def test_batch_inference(prepare_data):
        test_image, test_result = prepare_data
        batch_input = [test_image, test_image, test_image, test_image]
        expected_output = [test_result, test_result, test_result, test_result]
    
        model = InferenceModel()
        model.model_load(BACKEND)
        preprocessed_data = model.preprocess(batch_input)
        forwarded_data = model.forward(preprocessed_data)
        postprocessed_data = model.postprocess(forwarded_data)
    
        results = [[pydantic.json.pydantic_encoder(item) for item in image_result]
                   for image_result in postprocessed_data]
>       assert is_equal(results, expected_output)
E       AssertionError: assert False
E        +  where False = is_equal([[{'class_name': 'potted plant', 'score': 0.9737666249275208, 'x1': 363, 'x2': 397, ...}, {'class_name': 'dining table...7942, 'x1': 405, 'x2': 433, ...}, {'class_name': 'bowl', 'score': 0.9188650846481323, 'x1': 563, 'x2': 585, ...}, ...]], [[{'class_name': 'potted plant', 'score': 0.9767851233482361, 'x1': 363, 'x2': 398, ...}, {'class_name': 'chair', 'sco...1': 404, 'x2': 433, ...}, {'class_name': 'potted plant', 'score': 0.9167777299880981, 'x1': 170, 'x2': 234, ...}, ...]])

tests/test_pytorch_fastercnn.py:50: AssertionError
=============================== warnings summary ===============================
tests/test_pytorch_fastercnn.py::test_process_sample
  /Users/lxdv/conda/envs/env/lib/python3.7/site-packages/torch/nn/functional.py:718: UserWarning: Named tensors and all their associated APIs are an experimental feature and subject to change. Please do not use them for anything important until they are released as stable. (Triggered internally at  ../c10/core/TensorImpl.h:1156.)
    return torch.max_pool2d(input, kernel_size, stride, padding, dilation, ceil_mode)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
FAILED tests/test_pytorch_fastercnn.py::test_process_sample - AssertionError:...
FAILED tests/test_pytorch_fastercnn.py::test_batch_inference - AssertionError...
=================== 2 failed, 1 passed, 1 warning in 42.21s ====================
```

Reply:
Any ideas why do the tests fail?

Reply:
Different order of results?

Reply:
Didn't take a look at this. Just wanted to say that we provide a manual that developers could think is not fully working